src/coordinates.rs

```diff
@@ -0,0 +1,146 @@
+//! Custom data types used in the Gerber format.
+
+use std::convert::{From, Into};
+
+
+/// The coordinate format specifies the number of integer and decimal places in
+/// a coordinate number. For example, the `24` format specifies 2 integer and 4
+/// decimal places. The number of decimal places must be 4, 5 or 6. The number
+/// of integer places must be not more than 6. Thus the longest representable
+/// coordinate number is `nnnnnn.nnnnnn`. The same format must be defined for X
+/// and Y. Signs in coordinates are allowed; the `+` sign is optional.
+#[derive(Debug, Copy, Clone)]
+pub struct CoordinateFormat(pub u8, pub u8);
+
+
+/// Decimals are a sequence of one or more digits with an optional decimal
+/// point optionally preceded by a ‘+’ or a ‘-’ sign. They must fit in an IEEE
+/// double.
+///
+/// The value is stored as a 64 bit integer with 9 decimal places.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct Decimal(i64);
+
+const DECIMAL_PLACES_CHARS: u8 = 9;
+const DECIMAL_PLACES: i64 = 1_000_000_000;
+
+impl From<f64> for Decimal {
+    fn from(val: f64) -> Decimal {
+        let integer = (val.trunc() as i64) * DECIMAL_PLACES;
+        let decimal = ((val - val.trunc()) * DECIMAL_PLACES as f64).trunc() as i64;
+        Decimal(integer + decimal)
+    }
+}
+
+impl Into<f64> for Decimal {
+    fn into(self) -> f64 {
+        (self.0 as f64) / DECIMAL_PLACES as f64
```

Comment:
What does `self.0` mean? Do you have a link to the Rust documentation which describes this feature?